mesonbuild/backend/ninjabackend.py

```diff
@@ -1709,7 +1709,7 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
             args.extend(rustc.get_linker_always_args())
 
         args += self.generate_basic_compiler_args(target, rustc, False)
-        args += ['--crate-name', target.name]
+        args += ['--crate-name', target.name.replace('-', '_')]
```

Comment:
This deserves a comment that we’re matching rustic behavior here. Otherwise this looks good to me, thinks!